[config] Allow prometheus to configure IPv6 by using the standard net host port joining (#6815)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- Convert Prometheus histogram buckets to non-cumulative otel histogram buckets in `go.opentelemetry.io/contrib/bridges/prometheus`. (#6685)
- Don't start spans that never end for filtered out gRPC stats handler in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc`. (#6695)
- Fix a possible nil dereference panic in `NewSDK` of `go.opentelemetry.io/contrib/config/v0.3.0`. (#6752)

> - Fix problem with using IPv6 for prometheus endpoint. (#6815)

Reviewer: ```suggestion
- Fix prometheus endpoint with an IPv6 address in `go.opentelemetry.io/contrib/config`. (#6815)
```
